ci_nxp_build_failure: twister build for frdm_mcxa577/mcxa577 fails in gen_kobject_list.py with "ConstType" object has no attribute "size"

### Describe the bug

`west twister` build-only runs for `tests/drivers/build_all/eeprom` fail on `frdm_mcxa577/mcxa577` on Zephyr `main` at commit `8409c1339bc491cb94ce7f7835c4fb8903a96e56`.

Both `drivers.eeprom.build` and `drivers.eeprom.emul.build` fail while generating `kobject_prebuilt_hash.gperf`. The failure comes from `scripts/build/gen_kobject_list.py` with:

`AttributeError: 'ConstType' object has no attribute 'size'`

### Regression

- [x] This is a regression.

commit bf871df8e3d02bdeedc69da8d86a948a35da6b88
Author: Peter Wang <chaoyi.wang@nxp.com>
Date:   Mon Jun 29 16:26:32 2026 +0800

    boards: arm: nxp: frdm_mcxa577: add I3C thermometer support

    Enable I3C0 on the FRDM-MCXA577 board and register the on-board
    NXP P3T1755 temperature sensor as an I3C device. The clock is
    sourced from FRO_LF_DIV with divider 1 (per the board SDK
    hardware_init reference). Pins P0_20 (SDA), P0_21 (SCL), and
    P0_2 (PUR) are configured via a new pinmux_i3c0 group (verified
    against the SDK BOARD_InitI3CPins implementation).

    Add the ambient-temp0 alias required by the thermometer sample,
    list i3c in the board's supported features, and add frdm_mcxa577
    to the sample.sensor.thermometer_i3c integration platforms.

    Tested by building samples/sensor/thermometer.

    Signed-off-by: Peter Wang <chaoyi.wang@nxp.com>

 boards/nxp/frdm_mcxa577/board.c                   |  6 ++++++
 boards/nxp/frdm_mcxa577/frdm_mcxa577-pinctrl.dtsi | 18 ++++++++++++++++++
 boards/nxp/frdm_mcxa577/frdm_mcxa577.dtsi         | 17 +++++++++++++++++
 boards/nxp/frdm_mcxa577/frdm_mcxa577.yaml         |  1 +
 samples/sensor/thermometer/sample.yaml            |  1 +


### Steps to reproduce


 ```sh
   git clone https://github.com/zephyrproject-rtos/zephyr -b main
   cd zephyr
   git checkout 8409c1339bc491cb94ce7f7835c4fb8903a96e56
   west twister -p frdm_mcxa577/mcxa577 -T tests/drivers/build_all/eeprom --build-only --inline-logs -j 12 -M all
```


### Relevant log output

```shell
FAILED: zephyr/kobject_prebuilt_hash.gperf
...
Traceback (most recent call last):
  File "/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/scripts/build/gen_kobject_list.py", line 1102, in <module>
    main()
  File "/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/scripts/build/gen_kobject_list.py", line 1065, in main
    objs = find_kobjects(elf, syms)
  File "/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/scripts/build/gen_kobject_list.py", line 655, in find_kobjects
    objs = type_obj.get_kobjects(addr)
  File "/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/scripts/build/gen_kobject_list.py", line 329, in get_kobjects
    return type_env[self.child_type].get_kobjects(addr)
  File "/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/scripts/build/gen_kobject_list.py", line 277, in get_kobjects
    objs.update(mt.get_kobjects(addr + (i * mt.size)))
AttributeError: 'ConstType' object has no attribute 'size'
ninja: build stopped: subcommand failed.
```

### Impact

Annoyance – Minor irritation; no significant impact on usability or functionality.

### Environment

OS: Linux (Jenkins)
Zephyr revision: 8409c1339bc491cb94ce7f7835c4fb8903a96e56
Zephyr version: v4.4.0-7823-g8409c1339bc4
Toolchain variant: zephyr/gnu
Zephyr SDK: 1.0.1
Python: 3.12.3
CMake: 3.28.3
West: 1.5.0
Board/target: frdm_mcxa577/mcxa577
Test path: tests/drivers/build_all/eeprom

### Additional Context

_No response_

--- a/scripts/build/gen_kobject_list.py
+++ b/scripts/build/gen_kobject_list.py
@@ -319,6 +319,13 @@ class ConstType:
     def __repr__(self):
         return f"<const {self.child_type}>"
 
+    @property
+    def size(self):
+        # Delegate to the underlying type so that arrays of const-qualified
+        # kernel objects (e.g. "const struct device foo[N]") can compute the
+        # per-element stride. Mirrors has_kobject()/get_kobjects() delegation.
+        return type_env[self.child_type].size
+
     def has_kobject(self):
         if self.child_type not in type_env:
             return False

